### PR TITLE
Feature/lp 122 refactor components

### PR DESCRIPTION
**Pull request by NomasChuy**
This is the link to the pending task:

https://devaptivist.atlassian.net/browse/LP-122

I refactored the code to join all the similar components to a single generic file, added the creators comment on every function, deleted the old files, and removed the @Preview functions.

### CODE REVIEW

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/Button.kt [hunk 1]
```diff
@@ -64,4 +75,141 @@ fun LpEditFloatingActionButton(
...
+ * @param badgeNumber to modify the number of notifications shown
+ * @param onClick high order function, to define button action
+ *
+ * @author Jose Miguel Garcia Reyes
+ */
+@ExperimentalMaterial3Api
+@Composable
+fun BadgeButton(
+    modifier: Modifier = Modifier,
+    onClick: () -> Unit = {},
```
Comment (carlosph816) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/Button.kt]: We need to move this param to the end of the function and remove "= {}"

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/Button.kt [hunk 2]
```diff
@@ -64,4 +75,141 @@ fun LpEditFloatingActionButton(
...
+            )
+        }
+
+        if (badgeNumber > 0) {
+            BadgedBox(modifier = modifier
+                .padding(horizontal = 15.dp, vertical = 7.dp)
+                .align(Alignment.BottomEnd), badge = {
+                Badge(
+                    modifier = Modifier
+                        .padding(1.dp)
```
Comment (carlosph816) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/Button.kt]: We need to add the name param for example (all = 1.dp)

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/Button.kt [hunk 3]
```diff
@@ -64,4 +75,141 @@ fun LpEditFloatingActionButton(
...
+            uri?.let { uriSelected ->
+                onFileSelected(uriSelected)
+                fileName = uriSelected.getFileName(contentResolver)
+            }
+        }
+    Row(
+        verticalAlignment = Alignment.CenterVertically,
+        modifier = modifier
+            .border(
+                border = BorderStroke(1.dp, MaterialTheme.colorScheme.onPrimary),
```
Comment (carlosph816) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/Button.kt]: Add name params

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/Button.kt [hunk 4]
```diff
@@ -1,16 +1,27 @@
...
+import androidx.compose.foundation.border
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.selection.selectable
 import androidx.compose.foundation.shape.RoundedCornerShape
 import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Notifications
 import androidx.compose.material.icons.outlined.Edit
 import androidx.compose.material3.*
-import androidx.compose.runtime.Composable
+import androidx.compose.runtime.*
```
Comment (carlosph816) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/Button.kt]: We can avoid the ".*"

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/Card.kt [hunk 5]
```diff
@@ -207,28 +215,143 @@ fun LpIconTextCard(
...
+    title: String,
+    description: String,
+    onClick: () -> Unit
+) {
+    Card(
+        shape = MaterialTheme.shapes.medium.copy(all = CornerSize(12.dp)),
+        colors = CardDefaults.cardColors(
+            containerColor = Drab,
+        ),
+        modifier = Modifier.padding(10.dp)
```
Comment (carlosph816) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/Card.kt]: Add name param.

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/Card.kt [hunk 6]
```diff
@@ -207,28 +215,143 @@ fun LpIconTextCard(
...
+        Card(
+            colors = CardDefaults.cardColors(
+                containerColor = Bianca,
+            ),
+            shape = MaterialTheme.shapes.medium.copy(all = CornerSize(11.dp)),
+            modifier = Modifier
+                .width(177.dp)
+                .height(245.dp)
+                .clickable(onClick = onClick)
+                .padding(1.2.dp)
```
Comment (carlosph816) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/Card.kt]: Add name param.

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/Card.kt [hunk 7]
```diff
@@ -207,28 +215,143 @@ fun LpIconTextCard(
...
+            Column {
+                Image(
+                    painter = rememberAsyncImagePainter(imageUrl),
+                    contentDescription = null,
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .height(169.dp),
+                    contentScale = ContentScale.Crop
+                )
+                Box(modifier = Modifier.padding(12.dp)) {
```
Comment (carlosph816) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/Card.kt]: Add name param.

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/Card.kt [hunk 8]
```diff
@@ -207,28 +215,143 @@ fun LpIconTextCard(
...
+                            modifier = Modifier.constrainAs(descriptionText) {
+                                top.linkTo(titleText.bottom)
+                            }
+                        )
+                    }
+                }
+            }
+        }
+    }
+}
```
Comment (carlosph816) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/Card.kt:348]: Add a space line in the end.

**Review by carlosph816 — COMMENTED** [review]
Waiting changes.

**Review by carlosph816 — APPROVED** [review]
Well done!

**Review by juanislas77 — APPROVED** [review]
Nice job!

**Pull request merged**